### DISCUSSION TITLE
Error 400 on inbox.get_messages()

### DISCUSSION THREAD

**Discussion by EdoElgx** [O365/python-o365#766] [Q&A]
Hi All,

I have the following code:
```
    credentials = (
        az_credentials['client_id'],
        az_credentials['secret']
    )
    az_account = Account(
        credentials,
        auth_flow_type='credentials',
        tenant_id=az_credentials['tenant_id'],
        main_resource='usersupport@andes-cs.com')
    if az_account.authenticate():
        log.info('Az Authenticated.')
    mailbox = az_account.mailbox()
    print(mailbox)      # MailBox from resource: usersupport@andes-cs.com
    inbox = mailbox.inbox_folder()
    print(inbox)      # Inbox from resource: usersupport@andes-cs.com
    messages = inbox.get_messages() # Error 400
    for msg in messages:
        print(msg)

```
When I query the Inbox, I get the following error messages:
`requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: https://graph.microsoft.com/v1.0/usersupport@andes-cs.com/mailFolders/Inbox/messages?%24top=25 | Error Message: Resource not found for the segment 'usersupport@andes-cs.com'.`

The URL seems ok to me.

Any ideas and hints are very welcomed.

Cheers.

**Comment by alejcas** [comment]
This is normally because you don't have permissions over the email account provided "usersupport@andes-cs.com"

**Reply by EdoElgx** [reply]
Thank you for your answer, Alejandro.

Indeed it seems related to permissions. But this App Account has administrator granted permissions to all accounts. In fact, I've tried the code with 4 different accounts and it works ok in all of them, except in this specific one.

Any idea where should I look?

Cheers.

**Reply by alejcas** [reply]
I don't have any clue... sorry

**Reply by azreal1231** [reply]
Hi did you come right with this?  atm im also getting this problem
im trying get get emails from an shared account and not having any luck.

**Comment by EdoElgx** [comment]
I couldn't.

I had to get the messages delivered to another mailbox which worked perfectly. Permissions were the same in both mailboxes.

Sorry.
